GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs

```diff
@@ -1700,138 +1698,138 @@ private AbstractSingleActor FindActiveOrNextDragonVoid(long time)
             {
-                AddSpreadSelectionDecoration(p, log, replay, spreadEffectsNM, 240, 5000);
+                AddTargetedExpulsionSelectionDecoration(p, log, replay, targetedExpulsionsNM, 240);
             }
-            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.HarvestTempleRedPuddleSelectCM, out IReadOnlyList<EffectEvent> redSelectedEffectsCM))
+            if (log.CombatData.TryGetEffectEventsByDstWithGUID(p.AgentItem, EffectGUIDs.HarvestTempleVoidPoolRedPuddleSelectionCM, out IReadOnlyList<EffectEvent> voidPoolsCM))
             {
-                AddVoidPoolSelectionDecoration(p, replay, redSelectedEffectsCM, 400);
+                AddVoidPoolSelectionDecoration(p, log, replay, voidPoolsCM, 400);
             }
-            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.HarvestTempleRedPuddleSelectNM, out IReadOnlyList<EffectEvent> redSelectedEffectsNM))
+            if (log.CombatData.TryGetEffectEventsByDstWithGUID(p.AgentItem, EffectGUIDs.HarvestTempleVoidPoolRedPuddleSelectionNM, out IReadOnlyList<EffectEvent> voidPoolsNM))
             {
-                AddVoidPoolSelectionDecoration(p, replay, redSelectedEffectsNM, 300);
+                AddVoidPoolSelectionDecoration(p, log, replay, voidPoolsNM, 300);
             }
         }
 
         /// <summary>
-        /// Void Pools - Red Pool AoE on selected players.
+        /// Void Pools - Red Pool AoE on selected players.<br></br>
+        /// Duration logged of 7936, overriding to 7000 for logs without Dynamic End Event.
         /// </summary>
+        /// <remarks>As of EVTC20241030 effects on players have Dynamic End Time.</remarks>
         /// <param name="p">Selected player.</param>
+        /// <param name="log">The log.</param>
         /// <param name="replay">Combat Replay.</param>
-        /// <param name="redSelectedEffects">Effects List.</param>
+        /// <param name="effects">Effects List.</param>
         /// <param name="radius">Radius of the AoE.</param>
-        private void AddVoidPoolSelectionDecoration(AbstractPlayer p, CombatReplay replay, IReadOnlyList<EffectEvent> redSelectedEffects, uint radius)
+        private void AddVoidPoolSelectionDecoration(AbstractPlayer p, ParsedEvtcLog log, CombatReplay replay, IReadOnlyList<EffectEvent> effects, uint radius)
         {
-            var redSelectedEffectsOnPlayer = redSelectedEffects.Where(x => x.Dst == p.AgentItem).ToList();
-            foreach (EffectEvent redSelectedEffect in redSelectedEffectsOnPlayer)
+            foreach (EffectEvent effect in effects)
             {
-                int duration = 7000;
-                int start = (int)redSelectedEffect.Time;
-                AbstractSingleActor dragonVoid = FindActiveOrNextDragonVoid(redSelectedEffect.Time);
+                long duration = 7000;
+                (long start, long end) lifespan = effect.HasDynamicEndTime ? effect.ComputeDynamicLifespan(log, 7936) : (effect.Time, effect.Time + duration);
```

Comment:
why the duration given to compute dynamic is different than the declared duration? Is the declared one wrong?

Reply:
7936 works correctly with dynamic end in normal mode, in challenge mode it doesn't and the aoe lingers on the player for those extra 936 milliseconds (in game you can see it too), so i'm using the fixed 7000 in that case